### PR TITLE
SLM-191: Email hint text and validation for Legal Senders on Contact Helpdesk page

### PR DESCRIPTION
**Pull request by nathanrussell-moj-digital**

### CODE REVIEW

File: server/routes/helpdesk/validators/emailValidator.test.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import validateEmail from './emailValidator'
 
 describe('emailValidator', () => {
-  Array.of('user@aardvark.com.cjsm.net', 'mailroom-user@hmpps.gov.uk').forEach(email => {
+  Array.of('user@aardvark.com', 'user@aardvark.com.notcjsm.net', 'mailroom-user@hmpps.gov.uk').forEach(email => {
```
Comment (nathanrussell-moj-digital) [inline server/routes/helpdesk/validators/emailValidator.test.ts:4]: `user@aardvark.com.notcjsm.net` - proves the implementation is marginally better than just "endsWith('cjsm.net')"

File: server/routes/helpdesk/validators/emailValidator.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 const EMAIL_PATTERN =
   /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+const CJSM_NET_EMAIL = /^.+[@.]cjsm\.net$/i
```
Comment (mikehalmamoj) [inline server/routes/helpdesk/validators/emailValidator.ts:3]: I didn't realise we had 2 email validators now...

**Review by mikehalmamoj — APPROVED** [review]

File: server/routes/helpdesk/validators/emailValidator.ts (hunk 2 above)
Comment (nathanrussell-moj-digital) [reply server/routes/helpdesk/validators/emailValidator.ts:3]: Yeah, they are annoyingly different :(
This one validates you have entered an email address and that its not a CJSM address
The other one you are thinking of is the request a link validator. It validates you have entered an email address (any email address), and the validation to assert it is a CJSM address is at the API

Even if we could make the logic the same and bring them into the same class/function; the resultant error message text will be different 😢

Comment (mikehalmamoj) [reply server/routes/helpdesk/validators/emailValidator.ts:3]: Don't these designers know anything about DRY? 🤣

**Pull request merged**